src/worker.cc

```diff
@@ -1111,6 +1146,16 @@ valhalla_exception_t::valhalla_exception_t(unsigned code, const std::string& ext
     message += ":" + extra;
 }
 
+// function to add warnings to proto info object
+void add_warning(valhalla::Api& api, unsigned code) {
+  auto message = warning_codes.find(code);
+  if (message != warning_codes.end()) {
+    auto* warning = api.mutable_info()->mutable_warnings()->Add();
+    warning->set_description(message->second);
+    warning->set_code(message->first);
+  }
+}
+
```

Comment:
we made a mistake when adding warnings to the response in #3588 : `add_warnings` wasn't in the `valhalla` namespace, so the linker didn't know where to find the implementation. it didn't hit us before bcs we were only ever using it in this file, but now we also use it in other files.